src/lib/Err.hs

```diff
@@ -42,61 +54,142 @@ data ErrType = NoErr
              | ZipErr
              | EscapedNameErr
              | ModuleImportErr
+             | MonadFailErr
                deriving (Show, Eq)
 
-type Except = Either Err
+type SrcPosCtx  = Maybe SrcPos
+type SrcTextCtx = Maybe (Int, String) -- Int is the offset in the source file
+data ErrCtx = ErrCtx
+  { srcTextCtx :: SrcTextCtx
+  , srcPosCtx  :: SrcPosCtx
+  , messageCtx :: [String] }
```

Comment:
Why a list of strings?

Reply:
It's just to model what we had previously, where we added context with `addContext :: MonadErr m => String -> m a -> m a
` which caught and re-raised with the extra string appended to the end. This way we've reified the context in a data structure. For now it's just those same strings, but we can expand it to include other, more structured sorts of context.